### PR TITLE
Migrate JENKINS_USE_LOCAL_BINARIES to --extract=local

### PR DESCRIPTION
**Pull request by fejta**
/assign @krzyzacy @ixdy 

Also make `bootstrap.py --repo=k8s.io/kubernetes=master,1111,2222` work correctly
Add `kubernetes_e2e.py` unit test coverage for `--extract` and `--cluster`

ref https://github.com/kubernetes/test-infra/issues/2829

### CODE REVIEW

**Comment by krzyzacy** [comment]
please rebase

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -1842,6 +1843,15 @@ def GetRealBootstrapJob(self, job):
...
 
+    def testValidEnv(self):
+        for job, job_path in self.jobs:
+            with open(job_path) as fp:
+                data = fp.read()
+            if 'kops' in job:  # TODO(fejta): update this one too
+                continue
+            self.assertNotIn('JENKINS_USE_LOCAL_BINARIES=', data, 'Send --extract=local to config.json, not JENKINS_USE_LOCAL_BINARIES in %s' % job)
+            self.assertNotIn('JENKINS_USE_EXISTING_BINARIES=', data, 'Send --extract=local to config.json, not JENKINS_USE_EXISTING_BINARIES in %s' % job)
+
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: seems the test below covers it?

File: jobs/move_extract.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
```
Comment (krzyzacy) [inline jobs/move_extract.py]: put in experiment? or do we need to use the script again?

**Comment by krzyzacy** [comment]
/lgtm

please rebase (again) and you'll need an new e2e image for this, maybe make one and commit with the change to avoid potential exploding?

File: jenkins/e2e-image/e2e-runner.sh [hunk 3]
```diff
@@ -74,10 +74,17 @@ fi
 
 
 # We get the Kubernetes tarballs unless we are going to use old ones
+if [[ -n "${RAW_EXTRACT:-}" ]]; then
+  echo 'RAW_EXTRACT is set, --extract set by $@'
+  # TODO(fejta): delete everything after here
 if [[ "${JENKINS_USE_EXISTING_BINARIES:-}" =~ ^[yY]$ ]]; then
```
Comment (ixdy) [inline jenkins/e2e-image/e2e-runner.sh]: I think this needs to be `elif` now?

File: jenkins/e2e-image/e2e-runner.sh [hunk 4]
```diff
@@ -74,10 +74,17 @@ fi
...
-  echo "Using existing binaries; not cleaning, fetching, or unpacking new ones."
+  echo 'ERROR: JENKINS_USE_EXISTING_BINARIES no longer supported'
+  echo 'Send --extract=none to scenarios/kubernetes_e2e.py'
```
Comment (ixdy) [inline jenkins/e2e-image/e2e-runner.sh]: does anything use `--extract=none`? it looks like kubetest doesn't even handle it right now?

File: jobs/move_extract.py (hunk 2 above)
Comment (fejta) [inline jobs/move_extract.py]: Yes, we need to use it to migrate `--extract=not-local` which includes every `ci-` job

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (fejta) [inline jenkins/bootstrap_test.py]: That one only checks .env files

File: jenkins/e2e-image/e2e-runner.sh (hunk 3 above)
Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]: Done

File: jenkins/e2e-image/e2e-runner.sh (hunk 4 above)
Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]: Nothing uses it (nothing uses JENKINS_USE_EXISTING_BINARIES either)

Kubetest doesn't need to, think about what this scenario means! 😉 

It is in kubernetes_e2e.py in order to distinguish "don't extract anything" from "haven't migrated from JENKINS_USE_FOO env to --extract yet"

**Comment by fejta** [comment]
Thanks, updated things so that `--extract=local` will also set `JENKINS_USE_LOCAL_BINARIES` since this will need to get submitted before I can update the image.

**Comment by fejta** [comment]
```
bash$ $GOPATH/src/k8s.io/test-infra/jenkins/bootstrap.py --job=pull-kubernetes-e2e-gce-etcd3 --repo=k8s.io/kubernetes=master,44245 --repo=k8s.io/release --service-account ~/robot.json --json

I0603 17:55:59.063] SUCCESS! -- 300 Passed | 0 Failed | 0 Pending | 343 Skipped 
I0603 18:01:50.474] PASS: pull-kubernetes-e2e-gce-etcd3
```

**Comment by fejta** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/44245/pull-kubernetes-e2e-gce-etcd3/34141/

```
I0603 18:06:44.234] Call:  /var/lib/jenkins/workspace/pull-kubernetes-e2e-gce-etcd3/./test-infra/jenkins/../scenarios/kubernetes_e2e.py --env-file=platforms/gce.env --env-file=jobs/pull-kubernetes-e2e.env --env-file=jobs/pull-kubernetes-e2e-gce-etcd3.env --extract=local --build --stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce-etcd3 --cluster= --timeout=55m


W0603 18:06:46.711] +(/workspace/e2e-runner.sh:125): main(): kubetest -v --dump=/workspace/_artifacts --extract=local --test '--test_args=--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]' --build --stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce-etcd3 --up --down --deployment=bash --timeout=55m

I0603 18:06:46.822] JENKINS_USE_LOCAL_BINARIES=y
```

**Pull request merged**